vllm/worker/neuron_model_runner.py

```diff
@@ -233,6 +247,39 @@ def prepare_model_input(
                                    sampling_metadata=sampling_metadata,
                                    multi_modal_kwargs=multi_modal_kwargs)
 
+    def _update_neuron_sampling_params(self, sampling_metadata):
+        # Update Neuron sampling parameters (GenerationConfig in Neuron)
+        current_sampling_params = self.model_config.sampling_params
+        assert current_sampling_params is not None, (
+            f"Failed to update sampling_params, "
+            f"current sampling params is {current_sampling_params}")
+
+        top_k = current_sampling_params.top_k.copy()
+        top_p = current_sampling_params.top_p.copy()
+        temperature = current_sampling_params.temperature.copy()
+        for index, sequence_group_to_sample in enumerate(
+                sampling_metadata.seq_groups):
+            top_k[index] = self._convert_to_neuron_top_k(
+                sequence_group_to_sample.sampling_params.top_k)
+            top_p[index] = sequence_group_to_sample.sampling_params.top_p
+            temperature[index] = \
+                sequence_group_to_sample.sampling_params.temperature
```

Comment:
We are only applying 3 out of couple of dozen sampling parameters available, it's definitely not a backward compatible change. Can we at least have a switch to turn it off? Would be also good to warn which parameters were not applied.

Reply:
hi @xendo , neuron will turn on the on device sampling as the default sampling approach to get higher perf. currently, the on-device sampling only supports top_k, top_p and temperature. we should provide a way for user to turn off the on-device sampling. @liangfu , do you think we should go with a environment variable like `NEURON_ON_DEVICE_SAMPLING_DISABLED`?

Reply:
added a switch to turn on-device sampling off.

Reply:
thanks!